### PR TITLE
Automatically comment Pull Request with the review checklist

### PR DESCRIPTION
**Pull request by frouioui**
## Description

WIP

## Related Issue(s)

- https://github.com/vitessio/website/issues/872

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [ ] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: .github/workflows/comment_pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            ### Review Checklist
+            
+            Hello reviewers! :wave: Please follow the checklist that follows when reviewing this Pull Request.
+            
+            #### General
+            - [ ] Ensure that the Pull Request has the correct `release notes` label.
+            - [ ] If a new flag is being introduced, review whether it is really needed. The flag names should be clear and intuitive (as far as possible), and the flag's help should be descriptive.
+  
+            #### Bug fixes
+            - [ ] There should be an unit or end-to-end tests.
```
Comment (deepthi) [inline .github/workflows/comment_pull_request.yml:33]:
```suggestion
            - [ ] There should be at least one unit or end-to-end test.
```

File: .github/workflows/comment_pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            - [ ] The Pull Request description should either include a link to an issue that describes the bug OR an actual description of the bug and how to reproduce, along with a description of the fix.
+            
+            #### Non-trivial changes
+            - [ ] There should be some comments as to why things are implemented the way they are.
```
Comment (deepthi) [inline .github/workflows/comment_pull_request.yml:37]:
```suggestion
            - [ ] There should be some code comments as to why things are implemented the way they are.
```

File: .github/workflows/comment_pull_request.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            
+            #### New/Existing features
+            - [ ] They should be documented, either by modifying the existing documentation or creating new documentation.
```
Comment (deepthi) [inline .github/workflows/comment_pull_request.yml:40]:
```suggestion
            - [ ] Should be documented, either by modifying the existing documentation or creating new documentation.
```

File: .github/workflows/comment_pull_request.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            - [ ] New features should have a link to a feature request issue or an RFC that documents the use cases, corner cases and test cases.
+            
+            #### Backward compatibility            
+            - [ ] Protobuf changes should be wire-compatible.
+            - [ ] Changes to `_vt` tables and RPC need to be backward compatibles.
```
Comment (deepthi) [inline .github/workflows/comment_pull_request.yml:45]:
```suggestion
            - [ ] Changes to `_vt` tables and RPCs need to be backward compatible.
```

**Review by deepthi — COMMENTED** [review]
Looks great! I have found some typos + some suggestions for improving the words :)

**Comment by deepthi** [comment]
> This PR will need to be backported to all the active release branches. The workflow needs to be present in the branch we want to merge in, otherwise, the review checklist will not be added to the newly created Pull Requests.

I don't know how important it is to back port this, but I'm not opposed to it.

**Comment by frouioui** [comment]
> > This PR will need to be backported to all the active release branches. The workflow needs to be present in the branch we want to merge in, otherwise, the review checklist will not be added to the newly created Pull Requests.
> 
> 
> 
> I don't know how important it is to back port this, but I'm not opposed to it.

If we do not backport it, the Review Checklist will not be commented when creating new PRs against release branches

**Review by deepthi — APPROVED** [review]
